### PR TITLE
fix: [CDS-83305]:adding index.yaml size logs if greater than 20mb

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Dumping index.yaml size logs if greater than 20mb(This size was discussed with Abhi and not been calculated through benchmarking).
## Testing
Screenshot 2023-11-21 at 12 03 37 AM

- [x] NG - HTTP Helm V3 logs in helm deployment
- [x] NG - HTTP Helm V2 logs in helm deployment
- [x] NG - S3 Helm V3 logs in helm deployment
- [x] NG - S3 Helm V2 logs in helm deployment
- [x] NG - HTTP Helm V3 logs in k8s deployment
- [x] NG - HTTP Helm V2 logs in k8s deployment
- [x] NG - S3 Helm V3 logs in k8s deployment
- [x] NG - S3 Helm V2 logs in k8s deployment
- [x] CG - S3 Helm V2 logs in helm deployment
- [x] CG - S3 Helm V3 logs in helm deployment 
(https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55518)

### CODE REVIEW

**Comment by TarunUba1** [comment]
trigger uts

**Comment by acasian-harness** [comment]
@TarunUba1 please describe the intention of the pr and what are the tests performed. Did we run with huge index.yaml, what are the cpu, memory use with the change you added. Thanks!

**Comment by TarunUba1** [comment]
> @TarunUba1 please describe the intention of the pr and what are the tests performed. Did we run with huge index.yaml, what are the cpu, memory use with the change you added. Thanks!

Hi @acasian-harness this change is done for sole purpose of dumping size of index.yaml logs if they are above a limit, the limit was discussed to be around 15-20mb in general. Proper benchmarking and cpu usages will be taken up by Abhi

**Comment by TarunUba1** [comment]
trigger uts

**Comment by TarunUba1** [comment]
trigger uts

**Comment by TarunUba1** [comment]
sonar lint again not picking up on tests

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmTaskHelperBase.java [hunk 1]
```diff
@@ -979,7 +984,6 @@ public void downloadChartFilesUsingChartMuseum(
   public void addChartMuseumRepo(String repoName, String repoDisplayName, int port, String chartDirectory,
       HelmVersion helmVersion, long timeoutInMillis, String cacheDir, HelmCommandFlag helmCommandFlag) {
     String repoAddCommand = getChartMuseumRepoAddCommand(repoName, port, chartDirectory, helmVersion, helmCommandFlag);
-
```
Comment (TarunUba1) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmTaskHelperBase.java:982]: trying adding the space back but futile

File: 970-api-services-beans/src/main/java/io/harness/helm/HelmConstants.java [hunk 2]
```diff
@@ -43,6 +43,8 @@ public final class HelmConstants {
   public static final String VALUES_YAML = "values.yaml";
   public static final String CHARTS_YAML_KEY = "Chart.yaml";
   public static final String CHART_VERSION = "${CHART_VERSION}";
+  public static final String INDEX_FILE_WARN_LOG =
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/helm/HelmConstants.java:46]: Should we log the index.yaml size as well?

Comment (TarunUba1) [reply 970-api-services-beans/src/main/java/io/harness/helm/HelmConstants.java:46]: we can, should i log repo name also then

**Comment by TarunUba1** [comment]
trigger uts

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger uts

**Review by abosii-harness — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger uts

**Pull request merged**